lib/ae_mdw/application.ex

```diff
@@ -32,6 +32,7 @@ defmodule AeMdw.Application do
       AeMdw.Sync.Watcher,
       AeMdw.Sync.Supervisor,
       AeMdw.Sync.AsyncTasks.Supervisor,
+      {Task.Supervisor, name: AeMdw.Sync.AsyncTasks.TaskSupervisor},
```

Comment:
We can instead add a `child_spec/1` function to the `...AsyncTasks.TaskSupervisor` so we don't have to duplicate the name of the process in two very different places

Reply:
In this case it's not doable because `...AsyncTasks.TaskSupervisor` is only the name (the module is the `Elixir.Task.Supervisor`). I have moved it to the `AsyncTasks.Supervisor` though in order to shutdown the task when a `AsyncTasks.Consumer` goes down..

Reply:
You can always define it like so:
```elixir
def child_spec(_args) do
  TaskSupervisor.child_spec(name: __MODULE__) # Or use whatever name you want
end
```
and then just do
```elixir
AeMdw.Sync.AsyncTasks.TaskSupervisor,
```
on the children list

Reply:
That is interesting but the `,,,AsyncTasks.TaskSupervisor` here does not exist, is just a name. 
Now the spec is private to `,,,AsyncTasks.Supervisor`. May we continue with it to avoid retesting after a creation of new module?

Reply:
ok